cmd/boulder/main.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2014 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	// Load both drivers to allow configuring either
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/mattn/go-sqlite3"
+
+	"github.com/letsencrypt/boulder/ca"
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/ra"
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/letsencrypt/boulder/va"
+	"github.com/letsencrypt/boulder/wfe"
+)
+
+func main() {
+	app := cmd.NewAppShell("boulder")
+	app.Action = func(c cmd.Config) {
+		// Create the components
+		wfe := wfe.NewWebFrontEndImpl()
+		sa, err := sa.NewSQLStorageAuthority(c.SA.DBDriver, c.SA.DBName)
+		cmd.FailOnError(err, "Unable to create SA")
+		err = sa.InitTables()
+		cmd.FailOnError(err, "Unable to initialize SA")
+		ra := ra.NewRegistrationAuthorityImpl()
+		va := va.NewValidationAuthorityImpl()
+		ca, err := ca.NewCertificateAuthorityImpl(c.CA.Server, c.CA.AuthKey, c.CA.Profile)
+		cmd.FailOnError(err, "Unable to create CA")
+
+		// Wire them up
+		wfe.RA = &ra
+		wfe.SA = sa
+		ra.CA = ca
+		ra.SA = sa
+		ra.VA = &va
+		va.RA = &ra
+		ca.SA = sa
+
+		// Go!
+		newRegPath := "/acme/new-reg"
```

Comment:
The "boulder" cmd is the monolithic executable. To de-dupe, that kind of setup needs to move into the wfe library instead of the cmd, I think.